modules/tensor_mechanics/src/materials/ComputeCappedWeakPlaneStress.C

```diff
@@ -0,0 +1,890 @@
+  else
+    _all_q[2].f = - p - _cstrength.value(intnl[1]);
+}
+
+void
+ComputeCappedWeakPlaneStress::dyieldFunctions(Real /*p*/, Real q, const std::vector<Real> & intnl)
+{
+  // derivatives wrt p
+  _all_q[0].df[0] = _tan_phi.value(intnl[0]);
+  _all_q[1].df[0] = 1.0;
+  _all_q[2].df[0] = - 1.0;
+
+  // derivatives wrt q
+  if (_small_smoother2 == 0.0)
+    _all_q[0].df[1] = 1.0;
+  else
+    _all_q[0].df[1] = q / std::sqrt(Utility::pow<2>(q) + _small_smoother2);
+  _all_q[1].df[1] = 0.0;
+  _all_q[2].df[1] = 0.0;
+}
+
+void
+ComputeCappedWeakPlaneStress::dyieldFunctions_di(Real p, Real /*q*/, const std::vector<Real> & intnl)
+{
+  // derivatives wrt intnl[0] (shear plastic strain)
+  _all_q[0].df_di[0] = p * _tan_phi.derivative(intnl[0]) - _cohesion.derivative(intnl[0]);
+  _all_q[1].df_di[0] = 0.0;
+  _all_q[2].df_di[0] = 0.0;
+  // derivatives wrt intnl[q] (tensile plastic strain)
+  _all_q[0].df_di[1] = 0.0;
+  _all_q[1].df_di[1] = - _tstrength.derivative(intnl[1]);
+  _all_q[2].df_di[1] = - _cstrength.derivative(intnl[1]);
+}
+
+void
+ComputeCappedWeakPlaneStress::dflowPotential(Real /*p*/, Real q, const std::vector<Real> & intnl)
+{
+  // derivatives wrt p
+  _all_q[0].dg[0] = _tan_psi.value(intnl[0]);
+  _all_q[1].dg[0] = 1.0;
+  _all_q[2].dg[0] = - 1.0;
+  // derivatives wrt q
+  if (_small_smoother2 == 0.0)
+    _all_q[0].dg[1] = 1.0;
+  else
+    _all_q[0].dg[1] = q / std::sqrt(Utility::pow<2>(q) + _small_smoother2);
+  _all_q[1].dg[1] = 0.0;
+  _all_q[2].dg[1] = 0.0;
+}
+
+void
+ComputeCappedWeakPlaneStress::d2flowPotential_di(Real /*p*/, Real /*q*/, const std::vector<Real> & intnl)
+{
+  // d(dg/dp)/dintnl[0]
+  _all_q[0].d2g_di[0][0] = _tan_psi.derivative(intnl[0]);
+  _all_q[1].d2g_di[0][0] = 0.0;
+  _all_q[2].d2g_di[0][0] = 0.0;
+  // d(dg/dp)/dintnl[1]
+  _all_q[0].d2g_di[0][1] = 0.0;
+  _all_q[1].d2g_di[0][1] = 0.0;
+  _all_q[2].d2g_di[0][1] = 0.0;
+  // d(dg/dq)/dintnl[0]
+  _all_q[0].d2g_di[1][0] = 0.0;
+  _all_q[1].d2g_di[1][0] = 0.0;
+  _all_q[2].d2g_di[1][0] = 0.0;
+  // d(dg/dq)/dintnl[1]
+  _all_q[0].d2g_di[1][1] = 0.0;
+  _all_q[1].d2g_di[1][1] = 0.0;
+  _all_q[2].d2g_di[1][1] = 0.0;
+}
+
+void
+ComputeCappedWeakPlaneStress::d2flowPotential(Real /*p*/, Real q, const std::vector<Real> & /*intnl*/)
+{
+  // d(dg/dp)/dp
+  _all_q[0].d2g[0][0] = 0.0;
+  _all_q[1].d2g[0][0] = 0.0;
+  _all_q[2].d2g[0][0] = 0.0;
+  // d(dg/dp)/dq
+  _all_q[0].d2g[0][1] = 0.0;
+  _all_q[1].d2g[0][1] = 0.0;
+  _all_q[2].d2g[0][1] = 0.0;
+  // d(dg/dq)/dp
+  _all_q[0].d2g[1][0] = 0.0;
+  _all_q[1].d2g[1][0] = 0.0;
+  _all_q[2].d2g[1][0] = 0.0;
+  // d(dg/dq)/dq
+  if (_small_smoother2 == 0.0)
+    _all_q[0].d2g[1][1] = 0.0;
+  else
+    _all_q[0].d2g[1][1] = _small_smoother2 / std::pow(Utility::pow<2>(q) + _small_smoother2, 1.5);
+  _all_q[1].d2g[1][1] = 0.0;
+  _all_q[2].d2g[1][1] = 0.0;
+}
+
+Real
+ComputeCappedWeakPlaneStress::yieldF(Real p, Real q, const std::vector<Real> & intnl) const
+{
+  std::vector<Real> yf(3);
```

Comment:
Heheheh, i'm not sure who "we" is in @dschwen's comment: 

> I thought we had become pretty good about ...

but it certainly isn't ME !   I'd never heard of std::array.  Thanks for teaching me.